src/main/java/io/jsonwebtoken/impl/crypto/EllipticCurveSignatureValidator.java

```diff
@@ -40,7 +40,10 @@ public boolean isValid(byte[] data, byte[] signature) {
         Signature sig = createSignatureInstance();
         PublicKey publicKey = (PublicKey) key;
         try {
-            return doVerify(sig, publicKey, data, signature);
+            int expectedSize = getSignatureByteArrayLength(alg);
+            //if the expected size is not valid for JOSE, fall back to ASN.1 DER signature
```

Comment:
Awesome - thanks for this backwards compatibility fix!  Can you please update the comment to be clear that this is for backwards compatible reasons only and not something that is normally part of the JWS specification?  We need to be clear about why we're doing this in case we forget in the future.